packages/theme-ui/src/core.js

```diff
@@ -1,52 +1,82 @@
-import styled from '@emotion/styled'
 import { ThemeContext as EmotionContext } from '@emotion/core'
 import { MDXProvider } from '@mdx-js/react'
 import { get } from '@styled-system/css'
 import merge from 'lodash.merge'
 import jsx from './jsx'
-import themed from './themed'
 import { Context, useThemeUI } from './context'
 import { useColorState } from './color-modes'
+import { createComponents } from './components'
 
-const createComponents = (components = {}) => {
-  const next = {}
-  Object.keys(components).forEach(key => {
-    next[key] = styled(components[key])(themed(key))
+/*
+ * changes
+ * - ~~ThemeProvider no longer merges outer context~~
+ * - ~~New NestedThemeProvider component (name tbd)~~
+ * - todo: nested provider does not accept components
+ */
+
+const applyColorMode = (theme, mode) => {
+  if (!mode) return theme
+  const modes = get(theme, 'colors.modes', {})
+  return merge({}, theme, {
+    colors: get(modes, mode, {})
   })
-  return next
 }
 
-export const ThemeProvider = ({
-  theme,
+export const RootProvider = ({
+  theme = {},
   components,
-  ...props
+  children,
 }) => {
+  // components are provided in the default Context
   const outer = useThemeUI()
-  const initialColorMode = outer.colorMode || (theme ? theme.initialColorMode : undefined)
-  const [ colorMode, setColorMode ] = useColorState(initialColorMode)
-  const context = merge({}, {
+  if (outer.colorMode) {
+    throw new Error(`
+      [ThemeProvider]: only one ThemeProvider component can be used at a time.
+    `)
+  }
+
+  const [ colorMode, setColorMode ] = useColorState(theme.initialColorMode)
+
+  const context = {
+    ...outer,
     colorMode,
     setColorMode,
-  }, outer, {
-    theme,
-    components: createComponents(components),
-  })
-
-  if (context.colorMode) {
-    const modes = get(context.theme, 'colors.modes', {})
-    context.theme = merge({}, context.theme, {
-      colors: get(modes, context.colorMode, context.theme.colors)
-    })
+    components: {
+      ...outer.components,
+      ...createComponents(components)
+    },
+    theme: applyColorMode(theme, colorMode),
   }
 
   return (
     jsx(EmotionContext.Provider, { value: context.theme },
       jsx(MDXProvider, { components: context.components },
         jsx(Context.Provider, {
           value: context,
-          children: props.children
+          children,
         })
       )
     )
   )
 }
+
+// todo: should this provide components too?
```

Comment:
It *might* make more sense to require using the official `MDXProvider` here instead, though that could make applying `theme.styles` a little trickier. Custom MDX components could be created with the Theme UI `Styled` components or add hooks for `theme.styles` using the `css` prop, which is more work...

I might keep the functionality but conditionally render the `MDXProvider` only *if* the `components` prop is truthy